Client/Application/AppDelegate.swift

```diff
@@ -81,6 +82,23 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         return true
     }
 
+    func application(application: UIApplication, openURL url: NSURL, sourceApplication: String?, annotation: AnyObject?) -> Bool {
+        if url.scheme == "firefox" {
+            if url.lastPathComponent == "open-url" {
+                let query = url.query
+                if query != nil {
+                    let index = query!.unescape().rangeOfString("=")
```

Comment:
same here. Nest this inside an `if let index = query!.unescape().rangeOfString("=") {...}`